pkg/controller/vsphere/actuator.go

```diff
@@ -72,12 +85,20 @@ func (a *Actuator) Create(ctx context.Context, machine *machinev1.Machine) error
 		return a.handleMachineError(machine, fmtErr, createEventAction)
 	}
 	if err := newReconciler(scope).create(); err != nil {
+		// save the taskRef in our cache in case of any error with patch.
+		if scope.providerStatus.TaskRef != "" {
+			a.taskIDCache[machine.Name] = scope.providerStatus.TaskRef
+		}
 		if err := scope.PatchMachine(); err != nil {
 			return err
 		}
 		fmtErr := fmt.Errorf(reconcilerFailFmt, machine.GetName(), createEventAction, err)
 		return a.handleMachineError(machine, fmtErr, createEventAction)
 	}
+	// save the taskRef in our cache in case of any error with patch.
+	if scope.providerStatus.TaskRef != "" {
+		a.taskIDCache[machine.Name] = scope.providerStatus.TaskRef
+	}
```

Comment:
Would it be better to break this up a bit and not duplicate the caching of the task ref?
```suggestion
	err := newReconciler(scope).create(); 
	// save the taskRef in our cache before checking the error in case of any errors later.
	if scope.providerStatus.TaskRef != "" {
		a.taskIDCache[machine.Name] = scope.providerStatus.TaskRef
	}
	if err != nil {
		if err := scope.PatchMachine(); err != nil {
			return err
		}
		fmtErr := fmt.Errorf(reconcilerFailFmt, machine.GetName(), createEventAction, err)
		return a.handleMachineError(machine, fmtErr, createEventAction)
	}
```